### PR TITLE
Fixed styling issues in Mobile view

### LINKED ISSUE #1570: Impl: Update layout for chats to display more information
### Problem

This is an implementation issue for implementing the design updates from #1523.

### Solution

Please implement the design/layout updates related to margin/padding mentioned in this comment: https://github.com/okTurtles/group-income/issues/1523#issuecomment-1532703769

Also, this [Mobile] update for how the chat looks: https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?type=design&node-id=26056%3A283397&t=WWXBArWhx0WmOPmp-1 - removes all the borders in chat mode to give more space

### LINKED ISSUE #1572: [Mobile] Notifications header position wrong
### Problem

notification-margin

1. The position of the heading is wrong
2. The margins/padding on the list border are wrong

### Solution

Match the design:

Screen Shot 2023-05-04 at 8 50 50 PM

### LINKED ISSUE #1574: [Mobile] Reaction in chat broken
### Problem

On Safari iOS. When clicking on a reaction:

broken-reaction 1

This is what happens:

broken-reaction

### Solution

Fix.

### LINKED ISSUE #1578: [Mobile] Welcome page not centered
### Problem

Welcome page is not horizontally centered

image

### Solution

Fix to be horizontally centered on mobile.

### LINKED ISSUE #1586: "Oops, you entered 0 or a negative number" for non-number values
### Problem

1. When creating a new group, on step 3, if one enters a non-numeric value (e.g., 'abc'), the error shown is 'Oops, you entered 0 or a negative number'.

### Solution

The error message should say that an invalid value was entered, maybe with something to the effect of 'only positive numbers are allowed'

### LINKED ISSUE #1588: Styling Issues in Mobile
### Problem

1. In Appearance Page
image
2. Padding in Menu
image
3. Wrong width of divider in Notifications page
image
4. Can not scroll in Modal and Tab (User Settings Modal)
### Solution
Fix these separate issues
- [x] In Appearance Page
- [x] Padding in Menu
- [x] Wrong width of divider in Notifications page
- [x] Can not scroll in Modal and Tab (User Settings Modal)

### PR DESCRIPTION
**Pull request by Silver-IT**
### Summary of Changes

- Closes #1570 
- Closes #1588 
- Closes #1572 
- Closes #1578 
- Closes #1574
- Closes #1586

### CODE REVIEW

File: frontend/views/components/menu/MenuItem.vue [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default ({
   inheritAttrs: false,
   methods: {
     handleSelect () {
-      this.$listeners.click && this.$listeners.click()
+      // this.$listeners.click && this.$listeners.click()
```
Comment (taoeffect) [inline frontend/views/components/menu/MenuItem.vue]: Why is this being commented out?

Which issue is this addressing?

File: frontend/views/containers/access/LoginForm.vue [hunk 2]
```diff
@@ -104,7 +112,13 @@ export default ({
...
       username: {
         [L('A username is required.')]: required,
-        [L('A username cannot contain whitespace.')]: noWhitespace
+        [L('A username cannot contain whitespace.')]: noWhitespace,
+        [L('A username can only contain letters, digits, hyphens or underscores.')]: allowedUsernameCharacters,
+        [L('A username cannot exceed {maxChars} characters.', { maxChars: usernameMaxChars })]: maxLength(usernameMaxChars),
+        [L('A username cannot contain uppercase letters.')]: noUppercase,
+        [L('A username cannot start or end with a hyphen.')]: noLeadingOrTrailingHyphen,
+        [L('A username cannot start or end with an underscore.')]: noLeadingOrTrailingUnderscore,
+        [L('A username cannot contain two consecutive hyphens or underscores.')]: noConsecutiveHyphensOrUnderscores
```
Comment (taoeffect) [inline frontend/views/containers/access/LoginForm.vue]: Copying these validations does not address #1525. Although the video @leihla uploaded might have shown the problem with the Login Form, that's not the issue, as the original problem happened on the SignUp form. The problem was that Leihla's LastPass extension was filling out the fields in a way that was not triggering the validations at all.

**Review by taoeffect — CHANGES_REQUESTED** [review]
Left some feedback.

@Silver-IT can you do #1525 in a separate PR please? That issue is more difficult to debug and address, and doing it in this PR will hold us up from being able to merge and address the other issues.

File: frontend/views/components/menu/MenuItem.vue (hunk 1 above)
Comment (Silver-IT) [reply frontend/views/components/menu/MenuItem.vue]: It's for the second Issue I pointed [here](https://github.com/okTurtles/group-income/issues/1574#issuecomment-1541196105). (_The other is that makeEmoticon event is emitted twice._)
Should I remove this comment?

**Comment by Silver-IT** [comment]
Sure. I will revert the changes for #1525 and make a new separate PR for it.

File: frontend/views/containers/access/LoginForm.vue (hunk 2 above)
Comment (Silver-IT) [reply frontend/views/containers/access/LoginForm.vue]: Just reverted.

**Review by taoeffect — CHANGES_REQUESTED** [review]
Great work @Silver-IT! Almost ready to merge, a few change requests:

I realize this wasn't on the list of requests, but I think the chat border looks a *lot* better with a rounded rectangle, can you give it a 5px `border-radius` like this?

Screen Shot 2023-05-18 at 5 29 04 PM

Also there are still some coloring issues with the background color of the message reaction. Please make the entire background of the popup a darker color (and the same color, so that it's the dark version of the popup that appears in Leihla's designs):

Screen Shot 2023-05-18 at 5 24 16 PM

Other than that, this PR is looking really good! 👍

**Comment by Silver-IT** [comment]
> I realize this wasn't on the list of requests, but I think the chat border looks a lot better with a rounded rectangle, can you give it a 5px border-radius like this?

Sure. To keep consistency, can we use `rem` instead of `px`? So it could be `0.3rem` which is `4.8px` or `0.25rem` which is `4px`.

**Comment by taoeffect** [comment]
> Sure. To keep consistency, can we use `rem` instead of `px`? So it could be `0.3rem` which is `4.8px` or `0.25rem` which is `4px`.

Sure 😄 👍

File: frontend/views/containers/proposals/Mincome.vue [hunk 3]
```diff
@@ -74,10 +74,10 @@ export default ({
     form: {
       mincomeAmount: {
         [L('This field is required')]: required,
-        [L('Oops, you entered 0 or a negative number')]: mincomePositive,
         [L('The amount must be a number. (E.g. 100.75)')]: function (value) {
           return currencies[this.groupSettings.mincomeCurrency].validate(value)
-        }
+        },
+        [L('Oops, you entered 0 or a negative number')]: mincomePositive
```
Comment (taoeffect) [inline frontend/views/containers/proposals/Mincome.vue]: While we're moving this around, can you change the text to read: `Mincome must be greater than 0`

File: frontend/views/containers/group-settings/GroupCreationModal.vue [hunk 4]
```diff
@@ -172,10 +172,10 @@ export default ({
       sharedValues: { },
       mincomeAmount: {
         [L('This field is required')]: required,
-        [L('Oops, you entered 0 or a negative number')]: mincomePositive,
         [L('The amount must be a number. (E.g. 100.75)')]: function (value) {
           return currencies[this.form.mincomeCurrency].validate(value)
-        }
+        },
+        [L('Oops, you entered 0 or a negative number')]: mincomePositive
```
Comment (taoeffect) [inline frontend/views/containers/group-settings/GroupCreationModal.vue]: While we're moving this around, can you change the text to read: `Mincome must be greater than 0`

File: frontend/views/components/menu/MenuContent.vue [hunk 5]
```diff
@@ -34,14 +44,16 @@ export default ({
...
   @include phone {
-    border-radius: $radius;
+    border-radius: $radius $radius 0 0;
     background-color: $background;
     position: relative;
     z-index: 2;
-    padding-top: 0.5rem;
-    padding-bottom: 1rem;
+    padding-bottom: 4rem;
+    min-height: unset;
```
Comment (taoeffect) [inline frontend/views/components/menu/MenuContent.vue:56]: I'm not sure if it's here or below in `&.c-responsive-menu`, but both you and @SebinSong addressed the menu padding (on the invite links) in different ways.

When `master` is merged, this results in a short menu:

a

I think Sebin changed a different file: https://github.com/okTurtles/group-income/pull/1596/commits/d021797191364ccdb58a535d45357d4e1297ba15#diff-7fe2ce36f43b01930f5123552fc91f518faae1c46755eb35ef7555a2d52a726fR463-R464 in his PR #1596.

So Alex, can you merge `master`, and the coordinate with @SebinSong to figure out the best way to address this issue between the two of your changes?

**Review by taoeffect — CHANGES_REQUESTED** [review]
Great work @Silver-IT!

I see that @SebinSong made some similar changes to address the menu padding in PR #1596 (see my comment below for details), and he also addressed some other issues I found in this PR. Can you merge `master` therefore and coordinate with him to figure out the best approach?

Regarding the chat view in mobile, since we push the chat view all the way to the edge of the screen, I think makes sense to remove the rounded border at on the container (not the SendArea, but the very outermost container):

Screenshot 2023-05-24 at 10 44 57 AM

**Review by taoeffect — CHANGES_REQUESTED** [review]
Oh! Found one more bug related to resizing the chat view:

https://github.com/okTurtles/group-income/assets/138706/76d3bbc6-3b61-4a4c-b0ef-1878c6d23c89

**Review by taoeffect — CHANGES_REQUESTED** [review]
On desktop and tablet we **do** want to keep the rounded corner at the bottom. Only on mobile should it not be rounded:

Screen Shot 2023-05-25 at 10 00 35

**Review by taoeffect — APPROVED** [review]
Great work @Silver-IT! 👏

**Pull request merged**